elastic/kibana
# [Stateful sidenav] Add space selector in header

In this PR I've added the space selector menu in the header of the new stateful navigation.

## Screenshot

<img width="421" alt="Screenshot 2024-04-24 at 15 25 32" src="https://github.com/elastic/kibana/assets/2854616/804f0073-bfc2-4bf3-90c5-6c7c5b1294e5">


## How to test

Add the following in the `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"

xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "solutionNavEnabled": true
```

* Create some spaces
* Verify that the space selector is in the header
